server/build_event_protocol/build_event_server/build_event_server.go

```diff
@@ -87,11 +87,18 @@ func (s *BuildEventProtocolServer) PublishBuildToolEventStream(stream pepb.Publi
 
 	eg, ctx := errgroup.WithContext(ctx)
 	forwardingStreams := make([]pepb.PublishBuildEvent_PublishBuildToolEventStreamClient, 0)
+	var closeStreamsOnce sync.Once
+	defer closeStreamsOnce.Do(func() { closeForwardingStreams(forwardingStreams) })
```

Comment:
moved this `defer` up here in case we ever support multiple forwarding streams in synchronous mode - an error in one stream would mean that we `return err`, so we need to clean up any other streams that got created